src/io_intf.ml

```diff
@@ -26,7 +26,11 @@ module type S = sig
 
   val rename : src:t -> dst:t -> unit
 
-  val append : t -> string -> unit
+  val append : ?auto:bool -> t -> string -> unit
```

Comment:
Perhaps something more explicit than just `auto` would be better, e.g. `auto_flush`?

Reply:
Or perhaps `val append_overcommit` as a separate endpoint. I don't like boolean arguments, and allocating an optional one each time feels gratuitous.